tests/test_filesystem.py

```diff
@@ -0,0 +1,83 @@
+import os
+import tempfile
+import unittest
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+import yaml
+
+from tests import utils
+
+
+@pytest.mark.filesystem
+class TestFileSystem(unittest.TestCase):
+    """
+    Test FileSystem operations.
+    """
+
+    def setUp(self):
+        self.cfg_file = tempfile.NamedTemporaryFile(mode="w", suffix=".cfg", delete=False)  # pylint: disable=R1732
+        self.test_username = "test_username"
+
+    def tearDown(self):
+        utils.clear_handlers_instances()
+        os.unlink(self.cfg_file.name)
```

Comment:
Why was it necessary to separate the tmp config file creation here instead of nesting all the tests steps under `with self.cfg_file as f:` ?

Reply:
I don't think it was necessary, but it made more sense initially as I wrote the file content in the setup in my initial version of the tests.
It doesn't seem to be relevant anymore since the tests don't use the same content and have to write their own.
I will move this to each test.